### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by MostafaGamal**

### CODE REVIEW

File: query.go [hunk 1]
```diff
@@ -772,6 +773,8 @@ func scanRow(cols []string, rows *sql.Rows) (Result, error) {
 		return nil, fmt.Errorf("Error scanning row: %s", err)
 	}
 
+	columnTypes, _ := rows.ColumnTypes()
```
Comment (mcherri) [inline query.go]: we should check for errors and return with the error like in line 773

File: query.go [hunk 2]
```diff
@@ -786,7 +789,17 @@ func scanRow(cols []string, rows *sql.Rows) (Result, error) {
 			case int:
 				result[cols[i]] = int64(v.(int))
 			case []byte: // text cols are given as bytes
-				result[cols[i]] = string(v.([]byte))
+				if columnTypes[i].DatabaseTypeName() == "JSONB" || columnTypes[i].DatabaseTypeName() == "JSON" {
+					var inter interface{}
+					err := json.Unmarshal(v.([]byte), &inter)
+					if err != nil {
+						result[cols[i]] = err.Error()
```
Comment (mcherri) [inline query.go]: If we have an error we should return it directly like line 773. We should not put the error in the result at all. Maybe we should return the string representation of the byte as in line 801 but for sure not the error.

**Review by mcherri — CHANGES_REQUESTED** [review]
Please fix

File: query.go (hunk 1 above)
Comment (MostafaGamal) [reply query.go]: FIXED

File: query.go (hunk 2 above)
Comment (MostafaGamal) [reply query.go]: FIXED

**Review by mcherri — APPROVED** [review]
:100:

**Pull request merged**